src/machines/java/com/enderio/machines/common/io/fluid/MachineFluidTank.java

```diff
@@ -1,272 +1,126 @@
 package com.enderio.machines.common.io.fluid;
 
-import com.enderio.EnderIO;
-import net.minecraft.world.InteractionHand;
-import net.minecraft.world.InteractionResult;
-import net.minecraft.world.entity.EquipmentSlot;
-import net.minecraft.world.entity.player.Player;
-import net.minecraft.world.item.BucketItem;
-import net.minecraft.world.item.ItemStack;
-import net.minecraft.world.item.Items;
-import net.minecraft.world.level.block.entity.BlockEntity;
-import net.minecraftforge.common.capabilities.ForgeCapabilities;
+import net.minecraft.nbt.CompoundTag;
 import net.minecraftforge.fluids.FluidStack;
-import net.minecraftforge.fluids.FluidType;
+import net.minecraftforge.fluids.IFluidTank;
 import net.minecraftforge.fluids.capability.IFluidHandler;
-import net.minecraftforge.fluids.capability.IFluidHandlerItem;
-import net.minecraftforge.fluids.capability.templates.FluidTank;
 import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
 
-import java.util.Optional;
-import java.util.function.Predicate;
+public class MachineFluidTank implements IFluidTank {
 
-public class MachineFluidTank extends FluidTank {
-    public boolean allowInput = true;
-    public boolean allowOutput = true;
+    public static final String Capacity = "Capacity";
```

Comment:
should we even save the Capacity? It should be resurrected from the FluidLayout imo

Reply:
Yeah, it is required when the fluid tank data is read from NBT. Constructing the tank with fluid and not the capacity can be annoying, given TankLayout and NBT deserialization happens in 2 different places.  

Although I agree there is a bit of duplication, but I am not sure how to fix it .